src/SportsHub.Business/Repositories/ILanguageRepository.cs

```diff
@@ -0,0 +1,27 @@
+﻿using SportsHub.Shared.Entities;
+
+namespace SportsHub.Business.Repositories
+{
+    public interface ILanguageRepository
+    {
+        Task<IEnumerable<Language>> GetLanguagesAsync();
+
+        Task<Language> GetLanguageByIdAsync(Guid id);
+
+        Task<Language> GetLanguageByNameAsync(string languageName);
+
+        Task<Language> GetLanguageByCodeAsync(string languageCode);
+
+        Task AddLanguageAsync(Language language);
+
+        Task EditLanguageAsync(Language language);
+
+        Task<Language> DeleteLanguageAsync(Guid id);
+
+        Task<bool> DoesLanguageAlreadyExistByNameAsync(string languageName);
+
+        Task<bool> DoesLanguageAlreadyExistByCodeAsync(string languageCode);
+
+        Task<bool> DoesLanguageAlreadyExistByIdAsync(Guid id);
```

Comment:
I can suggest you better naming for these methods, I suppose.
```suggestion
        Task<bool> ExistsByIdAsync(Guid id);
```

Reply:
But it's simply a suggestion, If you want to, you can apply this renaming in separate PR for all methods

Reply:
I would keep it as it is. The naming is tied to the entity with which it operates.